### PR TITLE
[lock] Az lock delete fix bug: az lock delete does not work on Microsoft.DocumentDB

### PR DESCRIPTION
**Pull request by qianwens**
https://github.com/Azure/azure-cli/issues/12131

Root cause: lock resource id validation did not handle the third level resource like databaseAccounts/qianwencos/apis/mongodb/databases/mytestdb which cause the validation failed

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -1779,9 +1779,15 @@ def _validate_lock_params_match_lock(
...
-            parent = (resource['type'] + '/' + resource['name'])
+            if resource.get('child_type_3', None) is None:
+                _resource_type = resource.get('child_type_1', None)
+                _resource_name = resource.get('child_name_1', None)
+                parent = (resource['type'] + '/' + resource['name'])
+            else:
+                _resource_type = resource.get('child_type_2', None)
+                _resource_name = resource.get('child_name_2', None)
+                parent = (resource['type'] + '/' + resource['name'] + '/' +
+                          resource['child_type_1'] + '/' + resource['child_name_1'])
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: Do we need to add tests for this change?

Comment (qianwens) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Sure, added test case

**Pull request merged**